### PR TITLE
WIP: feat(dal): Move resourceToPayload to intrinsics

### PR DESCRIPTION
**Pull request by stack72**
We use this func everywhere so we are moving it to be an intrinsic

### CODE REVIEW

**Comment by nickgerace** [comment]
bors merge

**Comment by nickgerace** [comment]
bors cancel

File: lib/dal/src/pkg/import.rs [hunk 1]
```diff
@@ -88,6 +88,8 @@ async fn import_change_set(
             || SPECIAL_CASE_FUNCS.contains(&func_spec.name())
             || func_spec.is_from_builtin().unwrap_or(false)
         {
+            dbg!("Is an intrinsic");
+            dbg!(&func_spec.name());
```
Comment (nickgerace) [inline lib/dal/src/pkg/import.rs:92]: We should probably remove these

File: lib/dal/src/pkg/import.rs [hunk 2]
```diff
@@ -59,7 +59,7 @@ pub struct ImportOptions {
     pub is_builtin: bool,
 }
 
-const SPECIAL_CASE_FUNCS: [&str; 2] = ["si:resourcePayloadToValue", "si:normalizeToArray"];
+const SPECIAL_CASE_FUNCS: [&str; 1] = ["si:normalizeToArray"];
```
Comment (nickgerace) [inline lib/dal/src/pkg/import.rs:62]: Hmmm maybe this should be an intrinsic too in another PR

**Comment by stack72** [comment]
Spoke with @zacharyhamm - we need to make some changes here to build a backend for this func a la https://github.com/systeminit/si/tree/main/lib/dal/src/func/backend
